woocommerce/woocommerce
# Add product form buttons and specific product edit page

### All Submissions:






### Changes proposed in this Pull Request:

https://user-images.githubusercontent.com/2240960/183488212-4146ab6e-4f0f-4d8a-bc72-9cffe68d86d1.mp4

The primary addition of this PR is a new edit screen for the new product page and the action buttons for the new form.

This did require numerous additional changes, one was the addition of the `Form` component and some changes in the data and component packages.

#34029  .

### How to test the changes in this Pull Request:

1. Load this branch and enable the `new-product-management-experience` feature flag by using the WCA Test Helper (Tools > WCA Test Helper > Features) and refresh the page.
2. Enable track logging by running this in your console: `localStorage.setItem( 'debug', 'wc-admin:*' );`
3. Navigate to **Products > Add New (MVP)** and type something in the name field
4. Click **Save draft** and a `wcadmin_product_edit` track should fire with some of the product details ( if a brand new product then these product details will be `undefined`).
5. Notice how a notice popped up with **Product successfully created** and the `path` in the url also changed to the new product id.
6. Edit the name again and click **Save draft** again, it should trigger a track again, this time with the product details defined.
7. Click **Preview** it should open a new tab with the product preview and the original tab should output a `wcadmin_product_preview_changes` track.
8. Click **Publish**, which should output a `wcadmin_product_update` track and also show a notice again **Product published.** with a **View in store** link that should open a new tab with a preview and also trigger the `wcadmin_product_preview_changes` track.
After publish the **Publish** button should also change to **Update**, the **Save draft** button should switch to **Switch to draft**
9. Click the dropdown beside the Update/Publish button and it should contain 3 options. Try each, they each should output their own tracks: `wcadmin_product_publish_and_copy`, `wcadmin_product_copy`, and `wcadmin_product_delete`
The two copy actions should create a new product with the `- Copy` tagged on to the name and redirected to it's edit screen (in draft).
10. When on an edit screen of a product click **Add New (MVP)** again, it should reset the form to pristine conditions again.

### Other information:




### FOR PR REVIEWER ONLY:
